coalib/parsing/TomlConfParser.py

```diff
@@ -0,0 +1,255 @@
+                           False, adds the newly parsed data to the existing
+                           one (and overwrites already existing keys with the
+                           newly parsed values).
+        :return:           A dictionary with (lowercase) section names as keys
+                           and their Setting objects as values.
+        """
+
+        if os.path.isdir(input_data):
+            input_data = os.path.join(input_data, Constants.local_coafile_toml)
+
+        if overwrite:
+            self.__init_sections()
+
+        with open(input_data, 'r') as file:
+            self.data = tomlkit.parse(file.read())
+
+        self.data = self.data.body
+
+        for item in self.data:
+            self.generate_section(item, input_data)
+        return self.sections
+
+    def get_section(self, name, create_if_not_exists=False):
+        """
+        Returns or creates a section with given name
+
+        :param name: The name of the section
+        :param create_if_not_exists: create a section if it does not exist
+        :return: Section of given name
+        """
+        key = self.__refine_key(name)
+        sec = self.sections.get(key, None)
+        if sec is not None:
+            return sec
+
+        if not create_if_not_exists:
+            raise IndexError
+
+        retval = self.sections[key] = Section(str(name))
+        return retval
+
+    @staticmethod
+    def __refine_key(key):
+        return str(key).lower().strip()
+
+    def generate_section(self, item, origin):
+        """
+        Generates section
+
+        :param item: Configuration group
+        :param origin: The file from which the configuration originated
+        """
+
+        section_name = item[0]
+        section_content = item[1]
+        appends = []
+
+        # Handle Default section
+        if not isinstance(section_content, Table):
+            original_value = section_content
+            current_section = self.get_section('default', True)
+            logging.warning('A setting does not have a section.'
+                            'This is a deprecated feature please '
+                            'put this setting in a section defined'
+                            ' with `[<your-section-name]` in a '
+                            'configuration file.')
+
+            if section_name is None:
+                section_content = section_content.as_string()
+                section_name = '(' + 'comment' + str(self.__rand_helper) + ')'
+                self.__rand_helper += 1
+            self.create_setting(current_section,
+                                section_name,
+                                section_content,
+                                original_value,
+                                origin,
+                                False
+                                )
+            return
+
+        # Get the keys to be appended
+        if 'appends' in section_content.keys():
```

Comment:
Just remember these all name should be converted to lowercase when even if it is capitalise in the toml file.

Reply:
`__prepare_key` in Section.py seems to be handling it for us.